src/Core/Domain/State/QueryResult/EditableState.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\State\QueryResult;
+
+use PrestaShop\PrestaShop\Core\Domain\Country\ValueObject\CountryId;
+use PrestaShop\PrestaShop\Core\Domain\State\ValueObject\StateId;
+use PrestaShop\PrestaShop\Core\Domain\Zone\ValueObject\ZoneId;
+
+/**
+ * Transfers state data for editing
+ */
+class EditableState
+{
+    /**
+     * @var StateId
+     */
+    private $stateId;
+
+    /**
+     * @var int
+     */
+    private $countryId;
+
+    /**
+     * @var int
+     */
+    private $zoneId;
+
+    /**
+     * @var string
+     */
+    private $name;
```

Comment:
If im right, EditableObjects should have all properties optional except the identification (in this case is the stateId).
The point is if you want to update only one property, you don't need to fill the whole object. And in the handler it checks whether the property needs to be change or no (if its null or has a value).

Reply:
Looked how the other editable objects are made(EditableEmployee, EditableCustomer for example). See the same as here with not optional fields. Waiting for confirmation from core team how it should be.

Reply:
oh sorry, my bad, I had in mind Command for editing. Everything good here.